### PR TITLE
Add contributing guide

### LINKED ISSUE #387: Add a contributing guide to repository
Include a guide on how to make contributions to technical reports.

### PR DESCRIPTION
**Pull request by csarven**
This is an initial stab at having a contributing guide to technical reports in this repository. It needs massaging .. and your valuable feedback.

I request that we focus on working out the essentials in this PR and iron things out as we go. Prefer to merge this "sooner than later".

Closes #387 .

### CODE REVIEW

**Comment by elf-pavlik** [comment]
I think we should make a clear distinction between documents that are being worked on in this repository and documents which have a working copy in another repo and this one only includes read-only snapshots.

Each document links to its original Github repository and any issues and PRs should be made there.

IMO it might be safer to use [Git Submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) to avoid trying to edit those read-only snapshots.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+Read the [Solid Code of
+Conduct](https://github.com/solid/process/blob/main/code-of-conduct.md) and the
+[Positive Work Environment at W3C: Code of Ethics and Professional
+Conduct](https://www.w3.org/Consortium/cepc/) to keep our community approachable
+and respectable.
+
+
+## Contributions
+
+All substantive contributors must be members of the CG. It’s easy to [join the
```
Comment (RubenVerborgh) [inline CONTRIBUTING.md:25]: Maybe reverse that first sentence? "In order to be a substantive contributor of the spec, you must be a member of the CG."

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+CG](https://www.w3.org/community/solid/join) if you’d like to contribute. People
+agree to the [W3C Community License Agreement
+(CLA)](http://www.w3.org/community/about/agreements/cla/) upon joining the CG.
+
+The works in this repository use the [MIT
+license](https://github.com/solid/specification/blob/main/LICENSE.md).
+
+We accept different types of contributions, including some that don't require
+you to write a single line of code:
```
Comment (RubenVerborgh) [inline CONTRIBUTING.md:34]: Missing list, or misplaced colon?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+This section describes how to prepare the technical reports and how they'll be
+published on the Solid website.
+
+### General guidance
+
+* Content: the full content of the technical report must be human-readable with
+  HTML alone. Any CSS and JavaScript should not interfere with the accessibility
+  of the content.
+* Offline-friendly: there must not be any external dependencies (e.g., a network
```
Comment (RubenVerborgh) [inline CONTRIBUTING.md:175]:
```suggestion
* Offline-friendly: there must not be any mandatory external dependencies (e.g., a network
```

Comment (RubenVerborgh) [reply CONTRIBUTING.md:175]: (if that is what we mean)

**Review by RubenVerborgh — APPROVED** [review]
Great, I like it!

**Review by justinwb — APPROVED** [review]
This looks good. One (non-blocking) suggestion for CONTRIBUTING.md is to mention / point to the panels for people that may be interested in contributing to certain work streams that are meant to ultimately result in contributions to solid/specification.

**Comment by kjetilk** [comment]
I'm in general :+1: to this, but I feel it is worth mentioning that the minimal contribution to Solid may be a [https://github.com/solid/user-stories](user story). It should be noted that it should conform to the guideline form. But that means we should also start using the user stories submitted there in our milestoning process so that people feel it is worthwhile to submit that way.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,241 @@
+# Solid Technical Reports Contributing Guide
+
+Thank you for investing your time into contributing to the Solid project!
```
Comment (TallTed) [inline CONTRIBUTING.md:3]:
```suggestion
Thank you for investing your time in contribution to the Solid project!
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+[publishing technical reports](#publishing-technical-reports) are intended to
+be read-only snapshots of the CG work items.
+
+[Solid
+Panels](https://github.com/solid/specification/blob/main/README.md#solid-panels)
+focus on certain work streams, with an aim to propose technical reports for
+the Solid ecosystem.
+
+To build shared understanding, [user
+stories](https://github.com/solid/user-stories) can also be submitted.
```
Comment (kjetilk) [inline CONTRIBUTING.md:81]:
```suggestion
A low-barrier way to add concrete suggestions is to submit a [user
story](https://github.com/solid/user-stories). User stories are descriptions of desired features written in a special way, see the description there for details. User stories may then be used when formulating use cases and requirements.
```

Comment (kjetilk) [reply CONTRIBUTING.md:81]: I decided I needed to figure out how to make concrete suggestions, so there you go!

**Review by kjetilk — APPROVED** [review]
With these changes, I think we're good to go!

**Pull request merged**